pkg/upgrade/upgrader/jiva_upgrade.go

```diff
@@ -117,35 +118,67 @@ func getControllerPatchDetails(d *appsv1.Deployment) (
 	return patchDetails, nil
 }
 
-func getReplica(replicaLabel, namespace string) (*replicaDetails, error) {
-	replicaObj := &replicaDetails{}
-	deployObj, err := getDeployment(replicaLabel, namespace)
-	if err != nil {
-		return nil, errors.Wrapf(err, "failed to get replica deployment")
-	}
-	if deployObj.Name == "" {
-		return nil, errors.Errorf("missing deployment name for replica")
-	}
-	replicaObj.name = deployObj.Name
-	version, err := getOpenEBSVersion(deployObj)
+func getPatchDetailsForDeploy(pvName string, deployObj *appsv1.Deployment) (*replicaPatchDetails, error) {
+	patchDetails, err := getReplicaPatchDetails(deployObj)
 	if err != nil {
 		return nil, err
 	}
-	if (version != currentVersion) && (version != upgradeVersion) {
-		return nil, errors.Errorf(
-			"replica version %s is neither %s nor %s\n",
-			version,
-			currentVersion,
-			upgradeVersion,
-		)
+	patchDetails.UpgradeVersion = upgradeVersion
+	patchDetails.PVName = pvName
+	return patchDetails, nil
+}
+
+func getReplica(pvName, replicaLabel, namespace, openebsNamespace string) (*replicaDetails, error) {
```

Comment:
n_: just to be clear, the variable `namespace` can be renamed as `volumeNamespace`